csrc/device_lower/analysis/tensor_memory.cpp

```diff
@@ -72,22 +120,51 @@ TensorMemoryInfo computeTMemInfo(Fusion* fusion) {
     region.address->setMemoryType(MemoryType::Shared);
 
     // Assign each tensor in the region a whole 128 lanes and N columns.
-    region.num_columns = region.address->fusion()->zeroVal(DataType::UInt16);
+    region.num_columns = fusion->zeroVal();
```

Comment:
nit: Why was it previously using `DataType::UInt16` but now with `DataType::Index`?

Reply:
This is mostly a convenience change. In the past, we accumulate using uint16, then cast to uint32. Now we do accumulation in index type, then cast to uint16 and uint32. As required by PTX, the address type must be uint16, and allocation size be uint32, so we need some cast anyway. This is just a matter of when doing the cast is more convenient for me to write, no deep reasons.